From 4de8456bf6c7ae2a30f2f84e4dc25837be804734 Mon Sep 17 00:00:00 2001
Message-Id: <patch-13.17-4de8456bf6c-20210711T161212Z-avarab@gmail.com>
In-Reply-To: <cover-00.17-00000000000-20210711T161212Z-avarab@gmail.com>
References: <pull.1012.v5.git.git.1625684869.gitgitgadget@gmail.com>
	<cover-00.17-00000000000-20210711T161212Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sun, 11 Jul 2021 16:58:59 +0200
Subject: [PATCH v6? 13/17] refs API: make refs_resolve_refdup() not set errno
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Move refs_resolve_refdup() from the legacy refs_resolve_ref_unsafe()
to the new refs_resolve_ref_unsafe_with_errno(). I have read its
callers and determined that they don't care about errno being set.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 refs.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/refs.c b/refs.c
index 214f3f4f79f..45b51c0a532 100644
--- a/refs.c
+++ b/refs.c
@@ -287,9 +287,11 @@ char *refs_resolve_refdup(struct ref_store *refs,
 			  struct object_id *oid, int *flags)
 {
 	const char *result;
+	int ignore_errno;
 
-	result = refs_resolve_ref_unsafe(refs, refname, resolve_flags,
-					 oid, flags);
+	result = refs_resolve_ref_unsafe_with_errno(refs, refname,
+						    resolve_flags, oid, flags,
+						    &ignore_errno);
 	return xstrdup_or_null(result);
 }
 
-- 
2.32.0-dev

